### PR TITLE
[GSoC] Added PIDController class to control module.

### LINKED ISSUE #26163: Create a PID controller object
There are probably many different specialized transfer functions that are often used. For example, a PID controller is a common transfer function. We could have an object that the user passes the gains and the transfer function is created.

```python
import sympy as sm
import sympy.physics.control as cn

KD, KP, KI = sm.symbols('K_D, K_P, K_I', real=True)
s = sm.symbols('s')

Pcont = cn.TransferFunction(KP, 1, s)
Icont = cn.TransferFunction(KI, s, s)
Dcont = cn.TransferFunction(KD*s, 1, s)

PIDcont = cn.Parallel(Pcont, Icont, Dcont)
```

```
In [3]: PIDcont
Out[3]: Parallel(TransferFunction(K_P, 1, s), TransferFunction(K_I, s, s), TransferFunction(K_D*s, 1, s))

In [4]: PIDcont.doit()
Out[4]: TransferFunction(K_I + s*(K_D*s + K_P), s, s)
```

### PR DESCRIPTION
**Pull request by abhiphile**
**PIDController Class Added to Control Module**
- Introduced a new `PIDController` class in the control module.
   - The `PIDController` class allows for the creation of PID controllers with the following parameters:
        - `kp` : Proportional gain (defaults to `Symbol('kp')` if not provided).
        - `ki` : Integral gain (defaults to `Symbol('ki')` if not provided).
        - `kd` : Derivative gain (defaults to `Symbol('kd')` if not provided).
        - `tf` : Derivative filter time constant (defaults to 0 if not provided).
        - `var` : The complex frequency variable (defaults to `s` if not provided).

#### References to other Issues or PRs
Fixes: https://github.com/sympy/sympy/issues/26163

#### Release Notes

* physics.control
  * Introduced a `PIDController` class in the control module.

### CODE REVIEW

**Review by faze-geek — COMMENTED** [review]
Not sure if we should add the pid-controller image at `doc/src/modules/physics/control/pid-controller.png`, that is a place for our technical documentation `.rst` files.

**Review by faze-geek — COMMENTED** [review]
You can also showcase solving a good symbolic question on the **pid-controller** at `doc/src/guides/physics/control_problems.rst` using sympy. Maybe after this PR is merged.

File: sympy/physics/control/lti.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from typing import Type
-from sympy import Interval, numer, Rational, solveset
+from sympy import Interval, numer, Rational, solveset, sqrt, Intersection
```
Comment (faze-geek) [inline sympy/physics/control/lti.py:2]: Arrange new imports lexicographically after adding them.

File: sympy/physics/control/lti.py [hunk 2]
```diff
@@ -1313,6 +1353,115 @@ def to_expr(self):
...
+
+    Parameters
+    ==========
+
+    kp : sympifiable
+        Proportional gain. Defaults to 0 if not specified.
+    ki : sympifiable
+        Integral gain. Defaults to 0 if not specified.
+    kd : sympifiable
+        Derivative gain. Defaults to 0 if not specified.
```
Comment (faze-geek) [inline sympy/physics/control/lti.py:1375]: You can explicitly mention the possible inputs like `Number`, `Symbol`, `Expression` instead of sympifiable.

**Comment by moorepants** [comment]
> Not sure if we should add the pid-controller image at doc/src/modules/physics/control/pid-controller.png, that is a place for our technical documentation .rst files.

Also, do not commit binary image files. This PR needs to be reopened without the png or it needs to be removed, squashed, and force pushed.

**Comment by moorepants** [comment]
Make sure that comments are address from the prior attempts:

https://github.com/sympy/sympy/pull/26287
https://github.com/sympy/sympy/pull/26285
https://github.com/sympy/sympy/pull/26175

File: sympy/physics/control/lti.py [hunk 3]
```diff
@@ -1075,6 +1075,46 @@ def eval_frequency(self, other):
...
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> s = Symbol('s')
+        >>> tf1 = TransferFunction(1, s + 1, s)
+        >>> tf1.bandwidth()
+        1.00000000000000
```
Comment (moorepants) [inline sympy/physics/control/lti.py:1092]: What happens when you have a symbolic transferfunction:

```
 >>> tf1 = TransferFunction(a, s + a, s)
```

?

Comment (moorepants) [reply sympy/physics/control/lti.py:1092]: Also, for a symbolic `power_gain=Symbol('b')`?

File: sympy/physics/control/lti.py [hunk 4]
```diff
@@ -1075,6 +1075,46 @@ def eval_frequency(self, other):
...
+        >>> tf2 = TransferFunction(1, s - a, s)
+        >>> tf2.bandwidth()
+        1.0*a
+
+        Notes
+        =====
+
+        When using symbolic TransferFunction, make sure to define symbols along with
+        their signs properly.
+        """
```
Comment (moorepants) [inline sympy/physics/control/lti.py:1106]: Please cite a reference for this "bandwidth" calculation. I'm not immediately familiar with this and what the "power gain" is.

Also, this should be in a separate PR, as it is not related to PID.

File: sympy/physics/control/lti.py [hunk 5]
```diff
@@ -1313,6 +1353,115 @@ def to_expr(self):
...
 
 
+class PIDController(TransferFunction):
+    r"""
+    A class for representing PID (Proportional-Integral-Derivative) controllers in control systems.
+    The PIDController class is a subclass of TransferFunction, representing the controller's
+    transfer function in the Laplace domain. The arguments are ``kp``, ``ki``, ``kd``, ``tf``,
+    and ``var``, where ``kp``, ``ki``, and ``kd`` are the proportional, integral, and derivative gains respectively,
+    ``tf`` is the derivative filter time constant, which can be used to filter out the noise and ``var`` is the
+    complex variable used in the transfer function.
```
Comment (moorepants) [inline sympy/physics/control/lti.py:1363]: Line wrap to 79 characters.

File: sympy/physics/control/lti.py [hunk 6]
```diff
@@ -1313,6 +1353,115 @@ def to_expr(self):
...
+
+    .. image:: pid-controller.png
+
+    Parameters
+    ==========
+
+    kp : sympifiable
+        Proportional gain. Defaults to 0 if not specified.
```
Comment (moorepants) [inline sympy/physics/control/lti.py:1371]: I think a default of 1 is better, which would result in the PID doing nothing instead of killing the control loop.

Comment (moorepants) [reply sympy/physics/control/lti.py:1371]: Actually, the defaults here should be symbolic. `ki, kd, kp` should be symbols.

File: sympy/physics/control/lti.py [hunk 7]
```diff
@@ -1313,6 +1353,115 @@ def to_expr(self):
...
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.physics.control.lti import PIDController
+    >>> kp, ki, kd = symbols('kp ki kd')
+    >>> p1 = PIDController(kp, ki, kd)
+    >>> print(p1)
+    PIDController(kp, ki, kd, 0, s)
+    >>> p1.doit()
+    TransferFunction(kd*s + ki + kp*s, s, s)
```
Comment (moorepants) [inline sympy/physics/control/lti.py:1391]: kd*s/s = kd, ki/s, kp*s/s = kp

does not seem correct

**Comment by abhiphile** [comment]
> Not sure if we should add the pid-controller image at `doc/src/modules/physics/control/pid-controller.png`, that is a place for our technical documentation `.rst` files.

I've seen a similar thing in 
https://github.com/sympy/sympy/blob/958cc95aebc35ced70d870586e68d0468684c1a0/sympy/physics/continuum_mechanics/beam.py#L61
And the image file is located in `doc/src/modules/physics/continumm_mechanics/allowed-sign-conventions.png`. Using this as a reference I've put my image file in `doc/src/modules/physics/control/pid-controller.png`

**Review by Sc0rpi0n101 — APPROVED** [review]
